### PR TITLE
[confighttp] add max_concurrent_connections to HTTPServerSettings

### PR DESCRIPTION
**Pull request by timannguyen**
**Description:** 

Adding `max_concurrent_connections` to allow http server to limits http connection. this protects the server from being bombarded by tons of connection in-process and result in OOM.

telemetry is an optional metrics to instrument httpserver with opencensus. 

**Link to tracking Issue:** 

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27066

**Testing:** 

unit test

**Documentation:** 

updated confighttp README with `max_concurrent_connections` and `telemetry`

### CODE REVIEW

**Comment by timannguyen** [comment]
@dmitryax do you happen to know who has capacity to review this?

**Comment by splunkericl** [comment]
adding @atoulme as well. We want to introduce this for splunkhecreceiver but this will modify the httpserversettings for every component that uses it. Do you know who is the right POC to consult about this change?

File: config/confighttp/metrics.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	defaultMaxHttpConnectionsMetricName = metricPrefix + "http_max_connections"
+	defaultHttpConnectionsMetricName    = metricPrefix + "http_connections"
+)
+
+// HttpServerTelemetryConfig defines the telemetry configuration for HTTPServerSettings
+type HttpServerTelemetryConfig struct {
+	// Enabled is set to enable opencensus metrics
+	Enabled bool `mapstructure:"enabled"`
+	// ExtraAttributes is the extra attributes for metrics
+	ExtraAttributes map[string]string `mapstructure:"extra_attributes"`
```
Comment (dmitryax) [inline config/confighttp/metrics.go:24]: Why do you need this parameter? Collector provides a way to add extra resource attributes to the emitted metrics in the `service::telemetry::resource` config.

**Review by dmitryax — COMMENTED** [review]
Please break this PR down:
1. Limit config option
2. Telemetry (it has to be emitted in both opencensus and otel as all other metrics)

**Comment by dmitryax** [comment]
Let's hold off until discussion https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27066#issuecomment-1769740566 is resolved

**Comment by timannguyen** [comment]
it looks that [mem](https://github.com/open-telemetry/opentelemetry-collector/issues/8632) is what we are expecting. this would overall resolve the underlying issue. i've talked to @splunkericl. we're likely to close this out and use https://github.com/open-telemetry/opentelemetry-collector/issues/8632 when it is avail

**Pull request closed without merging**